### ISSUE TITLE
Release tarball

### ISSUE THREAD

**Issue by mike-burns** [glimpse-editor/Glimpse#242]
Perhaps I missed it, but is there a stable tarball for the 0.1.0 release? I see Flatpak and Windows but I'm looking for something for those of us who work with OS package managers: a release tarball with a checksum.

If it's there, then my request is to make it more clear on the downloads page; if not, can it exist please?

It looks like you're using autotools, so this should be the result of running `make dist`.

**Comment by Member1221** [comment]
The GitHub autogenerated tarballs are on the releases page
https://github.com/glimpse-editor/Glimpse/releases

We do not officially support distro-specific packages, so we haven't made any `make dist` tarballs.

**Comment by mike-burns** [comment]
> We do not currently officially support distro-specific packages, so we haven't made any `make dist` tarballs.

I was going to play around with what would be involved in making the OpenBSD package, but that'd (preferably) involve a `make dist`-style tarball: something where `./configure` works out of the box and where the checksum doesn't change as GitHub changes things (the autogenerated tarballs are surprisingly inconsistent).

But I'm not sure what "we do not currently officially support distro-specific packages" means; do you mean that me making an OpenBSD package is undesirable?

**Comment by TrechNex** [comment]
No, just that we specifically don't build, maintain or host them. We encourage the wider community to do that for us, and are open to merging any code changes they might need to facilitate that. (Snapcraft is supported that way, for example).

The reason we don't have a `make dist` tarball is we were only made aware that was a thing shortly before 0.1.0 released and there was no time to make the necessary updates. We could raise this as work ticket and then discuss a community if it is small enough to resolve in a bug-fix release or if it needs to be pushed out to 0.2.0.

**Comment by mike-burns** [comment]
Thank you for the explanation -- that makes total sense. I'd love to see a tarball for 0.2.0; is there something I can do to help that happen?

**Comment by TrechNex** [comment]
Possibly. Basically we would need to gain an understanding of what `make dist` so we know if there are Glimpse-specific changes we need to make to that. Also steps in `INSTALL.in` relate to the upstream release tarball so it would also be useful to know how that needs to be changed.

My own knowledge of Linux/BSD packaging is fairly minimal (I've `rpmbuild` before, but that's about it!) and it's proved to be one of the more time-consuming activities, so I've been grateful so far for the people who've come in to do that for our project :)

(Also, if you do need changes to support BSD at a later date I would be very enthusiastic about adding those. It's something people have asked us to support a few times!)

**Comment by mike-burns** [comment]
That's encouraging, thank you!

I cloned the repo to try a `make dist` myself (honestly, I want a full `make distcheck`) but ran into library incompatibilities (I'm one version of OpenBSD behind stable. My bad). I'll have to try again when it's not Sunday night.

Future me, or other OpenBSD users:

```
export AUTOMAKE_VERSION=1.14
export AUTOCONF_VERSION=2.69
export CXX=clang++
export CPPFLAGS="-I/usr/local/include -I/usr/X11R6/include"
export LDFLAGS="-L/usr/local/lib -L/usr/X11R6/lib" 
./autogen.sh
gmake distcheck
```

**Comment by TrechNex** [comment]
If it helps you can also find the "known good" dependency versions in `HACKING`.

**Comment by TrechNex** [comment]
Thank you for those changes. @bochecha reported on the Matrix channels that `make distcheck` may not work with the current upstream code either atm, so tagging this to #243 so we remember to check their bug tracker on this.

**Comment by mike-burns** [comment]
Thanks for the tip to look upstream; it seems people are working on `distcheck` there: https://gitlab.gnome.org/GNOME/gimp/merge_requests/173

I'm running into all sorts of `check` and `distcheck` issues in the `po` directory. One of them is that `/bin/true` is hardcoded in various spots (`configure.ac`, `po/Makefile.in.in`, `po-windows-installer/Makefile.in.in`), but it's spelled `/usr/bin/true` on OpenBSD (and, it seems, other OSes).

The fact is, if `make distcheck` works for you maintainers, then you'll be able to create a release tarball with confidence. It doesn't matter as much whether it works for me. So I'm going on punt on playing with `distcheck` for now.

**Comment by TrechNex** [comment]
For 0.1.2 I'm going to have to do this manually, but for 0.2.0 we will hopefully have an automated Jenkins job for it.

cc @chaomodus as they're doing the server provisioning these days :)

**Comment by TrechNex** [comment]
I had to make one last fix for 6d19844 to ensure `make dist` worked, but otherwise it went without a hitch for this release: https://github.com/glimpse-editor/Glimpse/releases/tag/v0.1.2
